src/lib/ledger_catchup/super_catchup.ml

```diff
@@ -1077,167 +1081,180 @@ let run_catchup ~logger ~trust_system ~verifier ~network ~frontier ~build_func
+                        ]
+                      "Validation error: external transition with state hash \
+                       $state_hash and its children were rejected for reason \
+                       $reason" ;
+                    Mina_metrics.(
+                      Counter.inc Rejected_blocks.no_common_ancestor
+                        (Float.of_int @@ (1 + List.length children_transitions)))) ;
+              List.iter forest ~f:(fun subtree ->
+                  Rose_tree.iter subtree ~f:(fun cached ->
+                      ( Cached.invalidate_with_failure cached
+                        : External_transition.Initial_validated.t
+                          Envelope.Incoming.t )
+                      |> ignore))
+          | Ok (root, state_hashes) ->
+              [%log' debug t.logger]
+                ~metadata:
+                  [ ("downloader", Downloader.to_yojson downloader)
+                  ; ( "node_states"
+                    , let s = Node.State.Enum.Table.create () in
+                      Hashtbl.iter t.nodes ~f:(fun node ->
+                          Hashtbl.incr s (Node.State.enum node.state)) ;
+                      `List
+                        (List.map (Hashtbl.to_alist s) ~f:(fun (k, v) ->
+                             `List [ Node.State.Enum.to_yojson k; `Int v ])) )
+                  ]
+                "before entering state machine.  node_states: $node_states" ;
+              let root =
+                match root with
+                | `Breadcrumb root ->
+                    (* If we hit this case we should probably remove the parent from the
+                        table and prune, although in theory that should be handled by
+                       the frontier calling [Full_catchup_tree.apply_diffs]. *)
+                    create_node ~downloader t (`Root root)
+                | `Node node ->
+                    (* TODO: Log what is going on with transition frontier. *)
+                    node
+              in
+              [%log debug]
+                ~metadata:[ ("n", `Int (List.length state_hashes)) ]
+                "Adding $n nodes" ;
+              (* if state_hashes is Ok, then we iterate through the forest and fold over state_hashes and call run_state_machine on each node.  order doesn't really matter because nodes called "out of order" will enter the `Wait_for_parent` state and begin running again when ready *)
+              List.iter forest
+                ~f:
+                  (Rose_tree.iter ~f:(fun c ->
                        let node =
-                         create_node t ~downloader (`Hash (h, l, parent))
+                         create_node ~downloader t (`Initial_validated c)
                        in
-                       don't_wait_for (run_state_machine node >>| ignore) ) ;
-                     (h, l))
-                 : State_hash.t * Length.t )))
+                       ignore
```

Comment:
don't_wait_for is generally preferred over ignore in this case, because it provides more direction about the purpose of the ignore call (they are functionally the same though).